platform/ontology/client.go

```diff
@@ -2,13 +2,59 @@ package ontology
 
 import (
 	"fmt"
+	"github.com/trustwallet/blockatlas/coin"
 	"github.com/trustwallet/blockatlas/pkg/blockatlas"
 )
 
 type Client struct {
 	blockatlas.Request
 }
 
+type blockListResponse struct {
+	Action  string            `json:"Action"`
+	Error   int               `json:"Error"`
+	Desc    string            `json:"Desc"`
+	Version string            `json:"Version"`
+	Result  []blockListResult `json:"Result"`
+}
+type blockListResult struct {
+	PrevBlock     string `json:"PrevBlock"`
+	TxnsRoot      string `json:"TxnsRoot"`
+	BlockTime     int    `json:"BlockTime"`
+	NextBlock     string `json:"NextBlock"`
+	BookKeeper    string `json:"BookKeeper"`
+	TxnNum        int    `json:"TxnNum"`
+	Height        int    `json:"Height"`
+	Hash          string `json:"Hash"`
+	ConsensusData string `json:"ConsensusData"`
+	BlockSize     int    `json:"BlockSize"`
+}
+
+type blockByNumberResponse struct {
```

Comment:
please, avoid nested structs, It's complex, hard to understand, and not so testable. Try to use only in a simple case

Reply:
and you don't need to parse the entire response object

Reply:
so, basically there is no difference to parse / unparse the entire object. Also it is your point about nested structs is opposite. It is - not complex, it is easier to understand, it is much more testable. Also I cannot find more simple case than that.